### PR TITLE
Add closest 2D points search algorithm

### PR DESCRIPTION
**Pull request by pgimalac**
Add closest 2D points search algorithm and tests.

### CODE REVIEW

**Review by Volpe95 — APPROVED** [review]

**Comment by pgimalac** [comment]
@imp2002 do you mind reviewing this PR to avoid it being closed automatically (I don't know if just writing this comment prevents it ?)

**Comment by imp2002** [comment]
ok, I will review it, can help resolve the confict?😋

**Comment by pgimalac** [comment]
I didn't notice the conflict :')
I'll merge (soon).

File: DIRECTORY.md [hunk 1]
```diff
@@ -30,6 +30,7 @@
   * Searching
     * [Binary Search](https://github.com/TheAlgorithms/Rust/blob/master/src/searching/binary_search.rs)
     * [Binary Search Recursive](https://github.com/TheAlgorithms/Rust/blob/master/src/searching/binary_search_recursive.rs)
+    * [Closest Points](https://github.com/TheAlgorithms/Rust/blob/master/src/searching/closest_points.rs)
```
Comment (siriak) [inline DIRECTORY.md:33]: @imp2002 I don't think we still need DIRECTORY.md, do we?

File: README.md [hunk 2]
```diff
@@ -70,6 +70,7 @@ These are for demonstration purposes only.
 
 - [x] [Linear](./src/searching/linear_search.rs)
 - [x] [Binary](./src/searching/binary_search.rs)
+- [x] [Closest pair of 2D points](./src/searching/closest_points.rs)
```
Comment (siriak) [inline README.md:73]: It's not a search algorithm (they search in an array). This is more like geometry or something...

**Review by siriak — CHANGES_REQUESTED** [review]

File: DIRECTORY.md (hunk 1 above)
Comment (imp2002) [reply DIRECTORY.md:33]: why? Is not need DIRECTORY.md or action?

File: README.md (hunk 2 above)
Comment (pgimalac) [reply README.md:73]: Do you want me to create a "geometry" section then ? (and move the file in a "geometry" folder)

Comment (pgimalac) [reply README.md:73]: I noticed convex hull (which is also a geometry algorithm) is in general, do you want me to use general ?

Comment (siriak) [reply README.md:73]: Yes, let's move both of them to Geometry

File: DIRECTORY.md (hunk 1 above)
Comment (siriak) [reply DIRECTORY.md:33]: I think we can remove both, but let's discuss that on The Algorithms Discord server
No need to do that change in this PR

File: README.md (hunk 2 above)
Comment (siriak) [reply README.md:73]: Please re-request my review once this change is made

**Review by siriak — APPROVED** [review]
LGTM, thanks!

**Pull request merged**